plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -275,6 +273,7 @@ public void testReadWholePartition()
                         .addCopies(new FileOperation(METADATA_JSON, INPUT_FILE_NEW_STREAM), 1)
```

Comment:
Please add tests when doing count(*) from the whole table.

Add a test involving a filter on a non partition column on a partitioned table.

Also important check the file system accesses for `min(C)`/`max(C)` .